Source/Public/G1TFormatConvert.h

```diff
@@ -0,0 +1,4550 @@
+﻿#pragma once
+
+#ifndef G1T_FORMAT_CONVERT_H
+#define G1T_FORMAT_CONVERT_H
+
+///////////////////////////////
+//   GENERAL SWIZZLE CODE    //
+///////////////////////////////
+
+// All swizzle code credit to Piken (DwayneR) https://github.com/fdwr
+// NOTE: Not all code is used here. A lot is for reference and debugging
+
+template <typename T, size_t N>
+constexpr size_t countof(const T(&a)[N])
+{
+    return N;// _countof(a);
+};
+
+enum class BlockAxis
+{
+    LinearX,
+    LinearY,
+    MortonX,
+    MortonY,
+};
+
+// A single swizzle item (axis and dimension length).
+struct CustomSwizzleItem
+{
+    
+    BlockAxis blockAxis;
+
+    // Block size in terms of multiples of the smaller previous block.
+    // So a 4x4 block would have a block size of 4 since there was no block before it,
+    // or rather the smallest block was a 1x1 pixel. An 8x8 block following that 4x4
+    // would have a size of 2.
+    // Negative block size means right-to-left flow.
+    int32_t blockSize;
+};
+
+uint32_t CustomSwizzle(uint32_t width, uint32_t height, uint32_t x, uint32_t y, CustomSwizzleItem const* items, size_t itemCount)
+{
+    uint32_t innerBlockWidth = 1;     // Accumulated block size in pixels
+    uint32_t innerBlockHeight = 1;
+    uint32_t outerBlockWidth = 1;     // Accumulated size in blocks
+    uint32_t outerBlockHeight = 1;
+    uint32_t innerPixelCount = 1;
+    uint32_t outerBlockCount = 1;
+    uint32_t reducedX = x;            // Coordinate after each reduction by a block
+    uint32_t reducedY = y;
+    uint32_t adjustedWidth = width;
+    uint32_t outputX = 0;             // Output coordinate in pixels
+    uint32_t outputY = 0;
+    uint32_t outputOffset = 0;
```

Comment:
I named these poorly, as they may be the output of the *function*, but from a *buffer* perspective, they are actually the swizzled *input* offset. For some of the other functions below, I used `sourceOffset` which is also confusing.

```suggestion
    uint32_t swizzledX = 0;             // Swizzled coordinate in pixels
    uint32_t swizzledY = 0;
    uint32_t swizzledOffset = 0;
```